app/scripts/components/follow.jsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import { Button } from 'react-bootstrap';
+
+import AuthorActions from '../actions/author';
+
+export default React.createClass({
+
+  onClick: function (evt) {
+    var action = evt.target.value;
+    switch (action) {
+      case 'Unfriend':
+      case 'Unfollow':
+        AuthorActions.unsubscribeFrom(this.props.author);
+        break;
+      case 'Follow':
+      case 'Friend':
+        AuthorActions.subscribeTo(this.props.author);
+        break;
+    }
+
+    // hack to rerender follow button rather than making the whole app rerender
+    // because the current user updates
+    setTimeout(function () {this.forceUpdate()}.bind(this), 50);
```

Comment:
You can add properties to the author that expose `hasFriend`, `follows` and `followedBy` as list objects that this component can subscribe to using `listenTo` or `connect`. Does that make sense?